dashboard/app/serializers/api/v1/teacher_feedback_serializer.rb

```diff
@@ -17,5 +17,11 @@
 #
 
 class Api::V1::TeacherFeedbackSerializer < ActiveModel::Serializer
-  attributes :id, :teacher_id, :student_id, :level_id, :comment, :created_at
+  attributes :id, :teacher_name, :student_id, :level_id, :comment, :created_at
+
+  private
+
+  def teacher_name
+    User.find_by_id(object.teacher_id).name
```

Comment:
use the existing [teacher association](https://github.com/code-dot-org/code-dot-org/blob/4f54043c00d8182d9375ad190b5ab2cbb2d1d3f1/dashboard/app/models/teacher_feedback.rb#L24):

```ruby
def teacher_name
  object.teacher.name
end
```

Reply:
Some additional notes: 

[find_by](https://apidock.com/rails/v4.0.2/ActiveRecord/FinderMethods/find_by) methods return nil if it's not found. Same with the association. 

[find(id)](https://apidock.com/rails/v4.0.2/ActiveRecord/FinderMethods/find) or a bang (!) find_by method ([find_by!](https://apidock.com/rails/v4.0.2/ActiveRecord/FinderMethods/find_by%21)(id: id) or `find_by_id!(id)` will raise `ActiveRecord::RecordNotFound`

Even though `teacher_id` is required, the `teacher` association could be nil if that teacher has been deleted. We can prevent an error and return nil teacher_name by using the safe navigation operator (`teacher&.name`), and we could clean up a teacher's feedback when the teacher is deleted by adding an association from user back to feedback with `dependent: :destroy`